sdk/eventhub/Azure.Messaging.EventHubs/src/Primitives/EventProcessor{TPartition}.cs

```diff
@@ -615,7 +621,7 @@ async Task performProcessing()
                                 // If the batch was successfully processed, capture the last event as the current starting position, in the
                                 // event that the consumer becomes invalid and needs to be replaced.
 
-                                lastEvent = eventBatch?.LastOrDefault();
+                                lastEvent = (eventBatch != null && eventBatch.Count > 0) ? eventBatch[eventBatch.Count - 1] : null;
```

Comment:
This reads worse, but FxCop didn't like us calling LINQ methods on a collection that had an indexer like this.  I'm happy to also add an internal LastOrDefault extension method over the actual type if folks think that reads better.

Reply:
I've no objections to the refactor.  I would assume that the LINQ call in this basic form has the appropriate type-specific optimizations, but this is only slightly ugly.